c2rust-transpile/tests/snapshots/snapshots__transpile@main.c.snap

```diff
@@ -0,0 +1,51 @@
+---
+source: c2rust-transpile/tests/snapshots.rs
+expression: cat tests/snapshots/main.rs
+input_file: c2rust-transpile/tests/snapshots/main.c
+---
+#![allow(
+    dead_code,
+    non_camel_case_types,
+    non_snake_case,
+    non_upper_case_globals,
+    unused_assignments,
+    unused_mut
+)]
+unsafe fn main_0(
+    mut argc: ::core::ffi::c_int,
+    mut argv: *mut *mut ::core::ffi::c_char,
+    mut envp: *mut *mut ::core::ffi::c_char,
+) -> ::core::ffi::c_int {
+    return 0 as ::core::ffi::c_int;
+}
+pub fn main() {
+    let mut args_strings: Vec<Vec<u8>> = ::std::env::args()
```

Comment:
Also, we can do this separately, but it's probably better to use `args_os()` here instead of `args()`, as otherwise we can only handle a limited UTF-8 subset of what the C program could.  One issue, though, is that we'd want to call [`OsString::into_encoded_bytes`](https://doc.rust-lang.org/std/ffi/struct.OsString.html#method.into_encoded_bytes), but that was only stabilized in Rust 1.74, and our generated nightly currently targets Rust 1.70.

Reply:
I'd prefer doing that in a separate PR. Is it acceptable to use unstable features in c2rust?

Reply:
Yeah, a separate PR for that would be good.  `c2rust-transpile` needs to build on `stable`, so we can't use unstable features there, but this would be an as-of-Rust 1.70 unstable (but as-of-Rust 1.74 stable) feature for the generated code.  Using unstable features in the generated code is okay, as we already do for a bunch of things.  This one is more minor, but we know it's been stabilized upstream already, so it should be fine.